lib/src/easy_localization_app.dart

```diff
@@ -58,25 +58,24 @@ class _EasyLocalizationLocale extends ChangeNotifier {
   _init(Locale fallbackLocale, List<Locale> supportedLocales) async {
     // Get Device Locale
     _osLocale = await _getDeviceLocale();
+    log('easy localization: Device locale ${_osLocale.toString()}');
     // If saved locale then get
     if (_savedLocale != null && this.saveLocale) {
-      locale = _savedLocale;
-      log('easy localization: Load saved locale ${_savedLocale.toString()}');
+      log('easy localization: Saved locale loaded ${_savedLocale.toString()}');
+      locale = _savedLocale;      
     } else {
-      locale = supportedLocales.firstWhere((locale) => _checkInitLocale(locale),
+      locale = supportedLocales.firstWhere((locale) => _checkInitLocale(locale, _osLocale),
           orElse: () => _getFallbackLocale(supportedLocales, fallbackLocale));
     }
-    //Set locale
-    if (Intl.defaultLocale == null) locale = _locale;
-    log('easy localization: Set locale ${this._locale.toString()}');
+    if (Intl.defaultLocale == null) locale = _locale;    
   }
 
-  bool _checkInitLocale(Locale locale) {
+  bool _checkInitLocale(Locale locale, Locale osLocale) {
```

Comment:
@Overman775 I think can used `_osLocale` direct coz is globle var and already `init _osLocale = await _getDeviceLocale();` ,  need to remove `Locale osLocale` param